AlfrescoKit/AlfrescoKit/View Controllers/Accounts View Controller/AKUserAccountListViewController.m

```diff
@@ -135,9 +136,12 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         }
         
         __weak typeof(self) weakSelf = self;
-        [self.loginService loginToAccount:selectedAccount networkIdentifier:nil completionBlock:^(BOOL successful, id<AlfrescoSession> session, NSError *error) {
+        __block AlfrescoRequest *request = nil;
+        request = [self.loginService loginToAccount:selectedAccount networkIdentifier:nil completionBlock:^(BOOL successful, id<AlfrescoSession> session, NSError *error) {
+            [weakSelf.delegate controller:weakSelf didCompleteRequest:request error:error];
             [weakSelf.delegate userAccountListViewController:weakSelf didLoginSuccessfully:successful toAccount:selectedAccount creatingSession:session error:error];
         }];
+        [self.delegate controller:self didStartRequest:request];
```

Comment:
didStartRequest would provide a Null value to the delegate implementation if it was called before a request object was created. If an error were to occur, the did complete request would be called with the error so that it could be handled by the user.

One possibility is to put checks around the start callback to ensure the request is still valid?